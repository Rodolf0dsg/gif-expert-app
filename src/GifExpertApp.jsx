import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['shingeky']);

  const onAddCategory = ( newCategory ) => {

    if ( categories.includes(newCategory) ) return;

    // console.log( newCategory )
    setCategories( category => [ newCategory, ...category ]);

  }

  return (
    <>
      <h1>GifExpertApp</h1>

      < AddCategory 
        // setCategories={ setCategories } 
        onNewCategory={ onAddCategory }
      />

        {
          categories.map( (category) => (
              <GifGrid 
                key={ category }
                category={ category }
              />
            )) 
        }

    </>
  );
}
