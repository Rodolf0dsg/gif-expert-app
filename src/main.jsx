import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';

import { GifExpertApp } from './GifExpertApp.jsx';


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>,
);
