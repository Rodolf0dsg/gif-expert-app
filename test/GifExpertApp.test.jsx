import { fireEvent, render, renderHook, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";
import { useState } from "react";

describe('Pruebas en GifExpertApp', () => {

    test('Match en el snapshot [opcional]', () => {

        const { container } = render(<GifExpertApp />);

        expect( container ).toMatchSnapshot();
        
    });

    test('Si busco una categoria la renderiza', () => {

        const { getByText } = render(<GifExpertApp />);

        const input = screen.getByRole('textbox');

        fireEvent.input( input , { target: { value: 'DBZ' } });
        const form = screen.getByRole('form');
        fireEvent.submit( form );

        expect( getByText('DBZ')).toBeTruthy()

    })

});

