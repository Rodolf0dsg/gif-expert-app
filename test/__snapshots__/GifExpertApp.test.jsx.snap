// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Match en el snapshot [opcional] 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar un Gif"
      type="text"
      value=""
    />
  </form>
  <h3>
    shingeky
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
